### DISCUSSION TITLE
HELP NEEDED: Understanding homebrews new formula policies

### DISCUSSION THREAD

**Discussion by omhmichaels** [Homebrew/discussions#2475] [Writing Formulae/Casks] [answered]
Hey homebrew team, I have had trouble understanding exactly what your maintainer is refering to. I have a formula that i originally made to be dependent on macos due to issues with the code using "go mod tidy" if installing on linux as well as other minor problems that need to be fix upstream for those builds. 

It was my understanding homebrew was a package manager for macos with linux support. I was also under the impression that formula did not need to be compatible for all supported architectures. Also i was under the impression that using resource blocks with a checksum was correct. Please let me know what i need to do to get this completed. It would also be great if homebrew documentation was updated to accurately reflect your maintainers rules. 

I had watched a video by your creator about how homebrew was created to make it easy for people to add packages. This process has been anything but and very frustrating due to contrast between the documentation and your maintainer. 

Heres the pull request:  https://github.com/Homebrew/homebrew-core/pull/89477

At this point i have no idea what exactly is needed to be done or why exactly the maintainer is saying things are in the docs when they are not. Please help

**Comment by iMichka** [comment, marked as answer]
Hi. We do support Linux since release 2.0 of Homebrew.
Since release 3.3.0 exactly, which was not so long ago, we also require packages to build on both macOS AND Linux.

There is continuous integration in place, which defines the list of supported platforms. And it needs to be green to be able to add a formula.

As the full Linux support is quite new and might be surprising, I added a note to our documentation here:
https://github.com/Homebrew/brew/pull/12444

There might be gaps in our documentation, improvements are welcome :)

This does not answer to all your questions but I hope this brings the necessary clarification for one of the topics.

**Reply by omhmichaels** [reply]
Thank you definitely brought clarification for one of the topics.
